openssl/openssl.rs

```diff
@@ -22,26 +22,26 @@ macro_rules! openssl_digest_benches {
 
 mod digest {
     openssl_digest_benches!(sha1, crypto_bench::SHA1_BLOCK_LEN,
-                            hash::Type::SHA1);
+                            hash::MessageDigest::sha1());
     openssl_digest_benches!(sha256, crypto_bench::SHA256_BLOCK_LEN,
-                            hash::Type::SHA256);
+                            hash::MessageDigest::sha256());
     openssl_digest_benches!(sha384, crypto_bench::SHA384_BLOCK_LEN,
-                            hash::Type::SHA384);
+                            hash::MessageDigest::sha384());
     openssl_digest_benches!(sha512, crypto_bench::SHA512_BLOCK_LEN,
-                            hash::Type::SHA512);
+                            hash::MessageDigest::sha512());
 }
 
 mod pbkdf2 {
     use crypto_bench;
     use openssl;
     use test;
 
-    pbkdf2_bench!(hmac_sha1, 20, out, {
-        let vec = openssl::crypto::pkcs5::pbkdf2_hmac_sha1(
-                    &crypto_bench::pbkdf2::PASSWORD_STR, &crypto_bench::pbkdf2::SALT,
-                    crypto_bench::pbkdf2::ITERATIONS as usize, out.len());
-        for i in 0..out.len() {
-            out[i] = vec[i];
-        }
+    pbkdf2_bench!(hmac_sha256, 20, out, {
```

Comment:
Switching from SHA-1 to SHA-256 should be done in a separate commit.

Should we be benchmarking all of SHA-{1, 256, 384, 512}? It seems like we should benchmark the same set of algorithms for each library.